cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -356,59 +386,251 @@ def parse_harvard_opinions(reporter, volume, make_searchable):
+
+        auth = op.find("author")
+        if auth is not None:
+            author_tag_str = titlecase(auth.text.strip(":"))
+            author_str = titlecase(
+                "".join(extract_judge_last_name(author_tag_str))
+            )
+        else:
+            author_str = ""
+            author_tag_str = ""
+
+        per_curiam = True if author_tag_str == "Per Curiam" else False
+        # If Per Curiam is True set author string to Per Curiam
+        if per_curiam:
+            author_str = "Per Curiam"
+
+        op_type = map_opinion_type(op.get("type"))
+        opinion_xml = str(op)
+        logger.info("Adding opinion for: %s", citation.base_citation())
+        op = Opinion(
+            cluster_id=cluster_id,
+            type=op_type,
+            author_str=author_str,
+            xml_harvard=opinion_xml,
+            per_curiam=per_curiam,
+            extracted_by_ocr=True,
+        )
+        # Don't index now; do so later if desired
+        op.save(index=False)
+        new_op_pks.append(op.pk)
+    return new_op_pks
+
+
+def get_opinion_content(cluster):
+    """Get the opinions content for a cluster object
+
+    :param cluster: Cluster ID for a set of opinions
+    :return: Combined opinion text
+    """
+    opinions = []
+    for op in Opinion.objects.filter(cluster_id=cluster.id):
+        if len(op.html_with_citations) > 1:
+            opinions.append(op.html_with_citations)
+        elif len(op.html_columbia) > 1:
+            opinions.append(op.html_columbia)
+        elif len(op.plain_text) > 1:
+            opinions.append(op.plain_text)
+        elif len(op.xml_harvard) > 1:
+            opinions.append(op.xml_harvard)
+    op = " ".join(opinions)
+    soup = BeautifulSoup(op, features="html.parser")
+    return soup.text
+
+
+def clean_docket_number(docket_number: str) -> str:
+    """Strip non-numeric content from docket numbers
+
+    :param docket_number: Case docket number
+    :return: A stripped down docket number.
+    """
+
+    docket_number = re.sub("Department.*", "", docket_number)
+    docket_number = re.sub("Nos?. ", "", docket_number)
+    return docket_number
+
+
+def clean_body_content(case_body: str) -> str:
+    """Strip all non alphanumeric characters
+
+    :param case_body: Opinion text
+    :return:Opinion text with only alphanumeric characters
+    """
+    soup = BeautifulSoup(case_body, "lxml")
+    return re.sub(r"[^[a-zA-Z]", "", soup.text)
+
+
+def match_based_text(
+    case_body: str, possible_cases: List, docket_number: str
+) -> Optional[OpinionCluster]:
+    """Compare CL text to Harvard content to establish duplicates
+
+    :param case_body: Harvard case body to compare
+    :param possible_cases: List of opinions to check against
+    :param docket_number: The docket number
+    :return: OpinionCluster or None
+    """
+    harvard_characters = clean_body_content(case_body)
+    for case in possible_cases:
```

Comment:
This could be simplified:

```python
diff = len(harvard_character) / len(cl_characters)
if 0.3 > diff > 3:
```

Reply:
this .... doesn't seem right.

Reply:
I think this code makes more sense- I didn't see it until the test failed.  But I think the example above always fails.
```
diff = len(harvard_characters) / len(cl_characters)
        if diff < 0.3 or diff > 3:
            continue
```